### ISSUE TITLE
Allow detaching weapon accessories

### ISSUE THREAD

**Issue by Johnny8-0** [chummer5a/chummer5a#3469]
When adding weapon accessories to a weapon slot the drop down menu removes other accessories that occupy the same slot.

I would like to be able to still buy accessories for a weapon even if the slot they occupy is already occupied. E.g. I equip my ARES Predator with Gas Vent 2, which occupies the barrel slot. If I go on a stealth mission I would like to be able to remove the Gas Vent and replace it with a Silencer, which also occupies the Barrel slot. I would simply keep the silencer in my inventory and then equip it using Armorer per the rules when needed. Perhaps an advanced menu system for wepaons sorting your accessories by the slot the occupy so you can pick and choose would be nice.

The ability to buy accessories without highlighting a weapon to buy them for. That way you can just buy accessories on the side and keep them in your inventory instead of not being able to buy them at all.

**Additional context**

**Comment by chummer5a** [comment]
Yeah, that's been a bit of a bugbear for a while. Not super hard to implement, just not supported at the moment.

**Comment by Johnny8-0** [comment]
Ok, thanks. Hope to see it in the future. Keep up the good work, I love Chummer!

Sent from Valhalla

On 5 Mar 2019, at 10.02, Chummer 5 Is Alive <notifications@github.com> wrote:

Yeah, that's been a bit of a bugbear for a while. Not super hard to implement, just not supported at the moment.

**Comment by Johnny8-0** [comment]
Perhaps also an option to filter on weapon accessories per slot in the shopping menu for a better overview of which slots take which mods. 

Sent from Valhalla

On 5 Mar 2019, at 10.02, Chummer 5 Is Alive <notifications@github.com> wrote:

Yeah, that's been a bit of a bugbear for a while. Not super hard to implement, just not supported at the moment.

**Comment by Teksura** [comment]
Just noting this functionality appears to be implemented through the feature that gave us an "Installed" box for weapon accessories. Though the issue remains open. This issue appears to be resolved.
